CollectionsInterface/Exercise6/src/Club.java

```diff
@@ -0,0 +1,132 @@
+        List <ClubMember> arrayMembers = new ArrayList<>(members.values());
+        Set <ClubMember> setMembers = new HashSet<>(members.values());
+        if(setMembers.size()!=arrayMembers.size()){
+            members.remove(member.getId(), member);
+            throw new ClubException("This member already exists.");
+        }
+    }
+    private Boolean isAlreadyVoted (UUID idVoterMember){
+        if(alreadyVoted.containsKey(idVoterMember)){
+            disqualifiedMembers.put(idVoterMember, true);
+            return true;
+        }else{
+            alreadyVoted.put(idVoterMember, true);
+            return false;
+        }
+    }
+    // I'm not sure if it's okay to make this method because I can directly put disqualifiedMembers.containsKey(id) in the ifclause of addvote
+    private Boolean isDisqualified (UUID idMemberVoted){
+        if(disqualifiedMembers.containsKey(idMemberVoted)){
+            return true;
+        }else
+            return false;
+    }
+    private void subtractVote(UUID idVoterMember){
+        if(votes.containsKey(idVoterMember)){
+            if(votes.get(idVoterMember) == 1)
+                votes.remove(idVoterMember);
+            else
+                votes.replace(idVoterMember, votes.get(idVoterMember) - 1);
+        }
+    }
+    private void addVote(UUID idMemberVoted, UUID idVoterMember) throws ClubException {
+        if(isDisqualified(idMemberVoted)){
+            throw new ClubException("The member you are trying to vote is not qualified to be voted because he tried to vote more than once.");
+        }
+        else{
+            if(members.containsKey(idMemberVoted)){
+                if(this.isAlreadyVoted(idVoterMember)){
+                    throw new ClubException("You have already voted, you will be disqualified from being voted from now on.");
+                }else{
+                   if(votes.containsKey(idMemberVoted))
+                        votes.replace(idMemberVoted, votes.get(idMemberVoted) + 1);
+                    else
+                        votes.put(idMemberVoted, 1);
+                }
+            }else
+                throw new ClubException("The member you are trying to vote does not exist.");
+        }
+    }
+    public void votePresident(ClubMember member, UUID enteredId, UUID idMemberVoted) throws ClubException {
+        if(!members.containsKey(member.getId())){
+            throw new ClubException("You are not a member of this club!");
+        }else{
+            if(member.getId()!=idMemberVoted){
+                this.addVote(idMemberVoted, member.getId());
+                if(enteredId==null || enteredId!=member.getId()){
+                    this.subtractVote(member.getId());
+                    throw new ClubException("You will have one less vote because you did not enter your id or you enter a wrong id.");
+                }
+            }else
+                throw new ClubException("You can not vote yourself!");
+        }
+    }
```

Comment:
```suggestion
    public void votePresident(ClubMember member, UUID enteredId, UUID idMemberVoted) throws ClubException {
        if(!members.containsKey(member.getId())){
            throw new ClubException("You are not a member of this club!");
        }
        if(member.getId()==idMemberVoted){
            throw new ClubException("You can not vote yourself!");
        }
        if(enteredId==null || enteredId!=member.getId()){
            throw new ClubException("You will have one less vote because you did not enter your id or you enter a wrong id.");
        }
        this.addVote(idMemberVoted, member.getId());
    }
```
Creo que ahi deberia funcionar igual, y mas legible